server/opts.go

```diff
@@ -1487,14 +1487,28 @@ func parseCluster(v interface{}, opts *Options, errors *[]error, warnings *[]err
 	return nil
 }
 
-func parseURLs(a []interface{}, typ string) (urls []*url.URL, errors []error) {
+func parseURLs(a []interface{}, typ string, warnings *[]error) (urls []*url.URL, errors []error) {
 	urls = make([]*url.URL, 0, len(a))
 	var lt token
 	defer convertPanicToErrorList(&lt, &errors)
 
+	dd := make(map[string]bool)
+
 	for _, u := range a {
 		tk, u := unwrapValue(u, &lt)
 		sURL := u.(string)
+		if dd[sURL] {
+			err := &configWarningErr{
+				field: sURL,
+				configErr: configErr{
+					token:  tk,
+					reason: "Duplicate entry detected",
```

Comment:
How about adding the type of connection to the reason? Then it would read:
```suggestion
					reason: fmt.Sprintf("Duplicate %s entry detected", typ),
```

```
Duplicate route entry detected
```